components/raftstore/src/store/peer.rs

```diff
@@ -3850,10 +3852,12 @@ where
         // Broadcast heartbeat to make sure followers commit the entries immediately.
         // It's only necessary to ping the target peer, but ping all for simplicity.
         self.raft_group.ping();
+
         let mut msg = eraftpb::Message::new();
         msg.set_to(peer.get_id());
         msg.set_msg_type(eraftpb::MessageType::MsgTransferLeader);
         msg.set_from(self.peer_id());
+        msg.set_index(self.get_store().entry_cache_first_index().unwrap_or(0));
```

Comment:
If there is a lagging peer which keeps the cache from being dropped. Then transfer leader would be slow to fetch so many entries. How about the index that has been replicted to quorum?

Reply:
> If there is a lagging peer which keeps the cache from being dropped. Then transfer leader would be slow to fetch so many entries. How about the index that has been replicted to quorum?

It has the timeout mechanism so that the transfer leader delay will be capped. Using the quorum's replicated index may defeat the purpose because the leader may still try to replicate logs to the lagging one if it's alive.

Reply:
Yes, but the leader can replicate logs to the lagging one by the previous async fetch mechanism. No need to fetch all at once when transferring leader.

Reply:
> How about the index that has been replicted to quorum?

I think this is not enough. Assume there are 3 instances A, B, C. A and B has latest logs, during rolling upgrade, when A or B is restarted, it needs to wait for C to catch up.

Reply:
reasonable